apps/snitch_core/lib/core/data/schema/card.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Snitch.Data.Schema.Card do
+  use Snitch.Data.Schema
+
+  alias Snitch.Data.Schema.{CardPayment, Address, User}
+
+  @type t :: %__MODULE__{}
+
+  schema "snitch_cards" do
+    field(:last_digits, :string)
+    field(:name_on_card, :string)
+    field(:year, :integer)
+    field(:month, :integer)
+    field(:card_name, :string)
+    field(:brand, :string)
+    field(:first_name, :string, virtual: true)
+    field(:last_name, :string, virtual: true)
+
+    belongs_to(:address, Address)
+    belongs_to(:user, User)
+
+    has_many(:card_payments, CardPayment)
+    timestamps()
+  end
+
+  @required_fields ~w(last_digits name_on_card brand user_id month year)a
+  @optional_fields ~w(card_name address_id)a
+
+  @current_year DateTime.utc_now() |> Map.fetch!(:year)
```

Comment:
These cannot be module attributes as their values will be compile-time constants. Please compute them at runtime:
```elixir
def changeset(card, params, :create) do
  %{year: current_year, month: current_month} = DateTime.utc_now()
  ...
end
```